Code/GraphMol/Wrap/Sgroup.cpp

```diff
@@ -0,0 +1,124 @@
+//
+//  Copyright (C) 2019 Greg Landrum
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#define NO_IMPORT_ARRAY
+#include <boost/python.hpp>
+#include <boost/python/suite/indexing/vector_indexing_suite.hpp>
+#include <string>
+
+// ours
+#include <GraphMol/RDKitBase.h>
+#include <GraphMol/Sgroup.h>
+#include "props.hpp"
+
+namespace python = boost::python;
+
+namespace RDKit {
+
+namespace {
+python::tuple getMolSGroups(ROMol &mol) {
+  python::list res;
+  std::vector<SGroup> &sgs = getSGroups(mol);
+  for (auto &sg : sgs) {
+    res.append(&sg);
+  }
+  return python::tuple(res);
+}
+void clearMolSGroups(ROMol &mol) {
+  std::vector<SGroup> &sgs = getSGroups(mol);

```

Comment:
That's private because we don't want people accessing/using SGroups using the methods of the molecule. Since the `getSGroups()` function can return a non-const vector, the method for clearing the list on the C++ side is: `getSGroups(non_const_mol).clear()`